src/decoder/lattice-simple-decoder.cc

```diff
@@ -564,8 +564,11 @@ void LatticeSimpleDecoder::ProcessNonemitting() {
   for (unordered_map<StateId, Token*>::iterator iter = cur_toks_.begin();
        iter != cur_toks_.end();
        ++iter) {
-    queue.push_back(iter->first);
-    best_cost = std::min(best_cost, iter->second->tot_cost);
+    StateId key = iter->first;
+    if (fst_.NumInputEpsilons(key) != 0) {
+      queue.push_back(key);
+      best_cost = std::min(best_cost, iter->second->tot_cost);
```

Comment:
the assignment of best_cost should not be inside the if-statement.  Check that in my original PR I did not make this mistake.

Reply:
Oh okay. Actually the PR did not have changes in the lattice-simple-decoder, so I figured the best_cost is only over those objects which are actually being pushed to the queue. Anyway, I'll make this change.

Reply:
Thanks.  The idea is to make the code equivalent to what it was before (but faster).  The best_cost is over all live tokens, which includes states that don't have epsilon transitions frm them.